# Mini Project 1 — Analysis Notebook

**Your name:**  Siqi Liu
**Dataset:**  Youth Risk Behavior Surveillance System
**Date:**  May 13, 2026

This notebook has four sections. Work through them in order. Each section has instructions and a code cell to fill in. Add markdown cells to explain your thinking as you go — that writing is part of the assignment.

When you're done, publish this notebook to your GitHub repository and submit the URL to Canvas.

In [2]:
# Setup — run this cell first
# If any package is missing, it will install automatically
import subprocess, sys

def install(pkg):
    subprocess.check_call([sys.executable, "-m", "pip", "install", pkg, "-q"])

for pkg in ["pandas", "plotly", "kaleido", "nbformat"]:
    try:
        __import__(pkg)
    except ImportError:
        print(f"Installing {pkg}...")
        install(pkg)

import pandas as pd
import plotly.express as px

print("Setup complete.")

Setup complete.


---

## Section 1 — Overview

Before writing any code, fill in this section. A good Overview tells anyone reading your notebook — including a future employer — what the analysis is about before they see a single chart.

**Dataset:** `clean_yrbs_2023.csv` — cleaned **Youth Risk Behavior Surveillance System (YRBS)**–style student survey extract (~20,000 adolescent responses, 110 variables), exported from `mp1_data.py` for this course. *(This is the file actually loaded below; it is not the Kaggle teen CSV.)*

**Why this dataset:** It connects **sleep, bullying, and digital/social media exposure** to **mental health and distress** items that schools and designers care about, using a standardized youth health survey format.

**Three analytical questions:**

1. How is social media use associated with adolescent mental health and feelings of hopelessness?
2. What is the relationship between cyberbullying and adolescent mental health outcomes?
3. How does sleep relate to social media use and mental health among teenagers?

**What a practitioner would do with these findings:** Educators, school health staff, and people who design teen-facing technology can use patterns like these to prioritize **sleep promotion**, **anti-cyberbullying support**, and **digital well-being** messaging—while remembering this snapshot cannot prove which intervention will work for an individual student.

---

## Section 2 — Data Profile

Load your dataset and get a basic picture of what's in it. Answer these questions in a markdown cell below your code:

- How many rows and columns does your dataset have?
- What does each column represent?
- Are there any obvious data quality issues (missing values, unexpected types, inconsistent formatting)?
- Which column or columns will your analysis focus on, and why?

In [9]:
# Load your dataset
# Replace 'your_dataset.csv' with your actual filename.
# The file should be in the same folder as this notebook.
# If you're loading from an API result, replace pd.read_csv() with the appropriate call.
#
# Example (app review dataset from class):
# df = pd.read_csv('app_reviews_demo.csv')

df = pd.read_csv('clean_yrbs_2023.csv')  # ← replace with your filename

print(df.shape)
df.head()

(20103, 110)


,weight,stratum,psu,Sex,Race,Age,Grade,SexOrientation,HispanicLatino,Height,...,ParentSubstanceUse,ParentMentalIllness,ParentIncarceration,SchoolConnectedness,ParentalMonitoring,UnfairDisciplineAtSchool,DifficultyConcentrating,EnglishProficiency,SexSexualContact,Transgender
0,0.8614,103.0,16294.0,Female,NaN,14.0,9.0,Gay or Lesbian,NaN,1.65,...,2.0,1.0,2.0,2.0,2.0,2.0,2.0,1.0,Never had sex,No
1,0.8920,103.0,16294.0,Male,White,15.0,9.0,Heterosexual,2.0,NaN,...,2.0,2.0,2.0,1.0,1.0,2.0,1.0,1.0,Never had sex,No
2,0.5081,103.0,16294.0,Male,White,16.0,11.0,Bisexual,2.0,1.68,...,2.0,2.0,1.0,1.0,1.0,2.0,2.0,1.0,Both,No
3,1.1759,103.0,16294.0,Female,White,17.0,10.0,Heterosexual,2.0,NaN,...,2.0,2.0,2.0,2.0,1.0,2.0,1.0,1.0,Males,No
4,0.8920,103.0,16294.0,Male,White,14.0,9.0,Heterosexual,2.0,1.85,...,2.0,2.0,2.0,1.0,1.0,2.0,2.0,1.0,Females,No


In [10]:
# Check column types and missing values
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20103 entries, 0 to 20102
Columns: 110 entries, weight to Transgender
dtypes: float64(105), object(5)
memory usage: 16.9+ MB


In [11]:
# Summary statistics for numeric columns
df.describe()

,weight,stratum,psu,Age,Grade,HispanicLatino,Height,Weight,SeatBeltUse,DrinkingDriver,...,Sunburn,AdultMeetingBasicNeeds,ParentSubstanceUse,ParentMentalIllness,ParentIncarceration,SchoolConnectedness,ParentalMonitoring,UnfairDisciplineAtSchool,DifficultyConcentrating,EnglishProficiency
count,20103.000000,20103.000000,20103.00000,20005.000000,19862.000000,19849.000000,17814.000000,17814.000000,15071.000000,19598.000000,...,10788.000000,15750.000000,13227.000000,13305.000000,13342.000000,11177.000000,10850.000000,10583.000000,10946.000000,10795.000000
mean,1.000000,228.370243,414571.68925,15.892677,10.355100,1.798630,1.696357,68.680140,0.435605,0.149199,...,0.543938,0.861905,1.710516,1.676362,1.823115,1.461752,1.167005,1.794860,1.519368,1.022696
std,1.270645,73.287806,235438.54837,1.218455,1.095383,0.401034,0.102717,18.466386,0.495852,0.356294,...,0.498089,0.345011,0.453540,0.467881,0.381586,0.498557,0.372997,0.403823,0.499648,0.148938
min,0.002300,101.000000,16294.00000,12.000000,9.000000,1.000000,1.320000,31.300000,0.000000,0.000000,...,0.000000,0.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000,1.000000
25%,0.153200,201.000000,214944.00000,15.000000,9.000000,2.000000,1.630000,55.790000,0.000000,0.000000,...,0.000000,1.000000,1.000000,1.000000,2.000000,1.000000,1.000000,2.000000,1.000000,1.000000
50%,0.414300,212.000000,401089.00000,16.000000,10.000000,2.000000,1.700000,64.410000,0.000000,0.000000,...,1.000000,1.000000,2.000000,2.000000,2.000000,1.000000,1.000000,2.000000,2.000000,1.000000
75%,1.359400,300.000000,573255.00000,17.000000,11.000000,2.000000,1.780000,77.110000,1.000000,0.000000,...,1.000000,1.000000,2.000000,2.000000,2.000000,2.000000,1.000000,2.000000,2.000000,1.000000
max,22.543700,300.000000,788430.00000,18.000000,12.000000,2.000000,2.030000,180.990000,1.000000,1.000000,...,1.000000,1.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000,2.000000


**Your data profile notes**

Each row in the csv file is one adolescent survey record with many YRBS-style behavior and health items in wide format.

**What’s in the columns**  
The first columns include **survey design fields** (`weight`, `stratum`, `psu`) used when analysts want **population-representative** estimates (not just raw row counts). Then come **demographics** (`Sex`, `Race`, `Age`, `Grade`, `SexOrientation`, `HispanicLatino`, `Height`, `Weight`) and a long tail of **risk and well-being items** (e.g. seat belt use, substance use, violence, bullying, `CyberBullying`, `Hopelessness`, sleep `EightOrMoreHoursSleep`, `SocialMedia`, school connectedness, parental factors, and suicide-related items). **`df.info()`** shows **105 numeric (`float64`)** columns and **5 object (`text`)** columns — so most answers were read in as numbers (often 0/1 or small ordinal codes), while a few stay as text categories.

**What I noticed from `df.describe()` (data quality)**  
`describe()` only summarizes **numeric** columns, but the **row counts per column** already tell a story: **`Age`** has **20,005** non-null values (about **98** missing out of 20,103), and many behavior columns have **even fewer** non-null responses. That pattern usually means **survey skip rules** or **optional modules**, not random typos. For my three questions (social media, cyberbullying, sleep/mental health), I need to **check missingness on those specific variables** before comparing groups, and I should not treat “missing” as the same as “no risk.”

**How this connects to my analysis plan**  
For **Question 1 (social media and mental health)**, I will focus on **`SocialMedia`** together with outcomes such as **`Hopelessness`** or **`NotGoodMentalHealth`**, and I will note whether heavy social media users are missing other fields more often. For **Question 2 (cyberbullying)**, **`CyberBullying`** is an obvious driver column, but I should relate it to outcomes without implying causation from a cross-section. For **Question 3 (sleep)**, **`EightOrMoreHoursSleep`** pairs with stress or mood items.

**Questions the profile raises**  
 The data is binary, so it did not tell the severity, just yes/no for each data entry. Also because this is **one time point per respondent**, any strong association is a **hypothesis for further study**, not proof that one behavior *caused* an outcome.

---

## Section 3 — Analysis

Answer your three research questions using pandas. Each question should have:

1. A markdown cell stating the question
2. A code cell with the analysis
3. A markdown cell with your interpretation — what does the result mean?

You may need to clean or reshape the data before you can answer a question. That's normal — document what you did and why.

**Question 1:** *How is social media use associated with adolescent mental health and feelings of hopelessness?*

In [20]:
# Question 1 — Social media indicator vs hopelessness (both coded 0/1 in this extract).
# I drop rows missing either answer so the rate compares the same kind of respondent.

q1 = df[["SocialMedia", "Hopelessness"]].dropna()
print(f"Students who answered BOTH items: {len(q1):,} (out of {len(df):,} total rows)\n")

tbl = q1.groupby("SocialMedia")["Hopelessness"].agg(["count", "mean"]).round(3)
tbl["pct_hopeless_1"] = (tbl["mean"] * 100).round(1)
print("Mean Hopelessness (= share answering 'yes' if 1 means yes) by SocialMedia code:")
print(tbl)

print("\nPlain-language rates (Hopelessness == 1):")
for code, grp in q1.groupby("SocialMedia"):
    pct = (grp["Hopelessness"] == 1).mean() * 100
    print(f"  SocialMedia == {int(code)}: {pct:.1f}% report hopelessness (n={len(grp):,})")

Students who answered BOTH items: 15,075 (out of 20,103 total rows)

Mean Hopelessness (= share answering 'yes' if 1 means yes) by SocialMedia code:
             count   mean  pct_hopeless_1
SocialMedia                              
0.0           3277  0.332            33.2
1.0          11798  0.441            44.1

Plain-language rates (Hopelessness == 1):
  SocialMedia == 0: 33.2% report hopelessness (n=3,277)
  SocialMedia == 1: 44.1% report hopelessness (n=11,798)


**Interpretation:** Among students who answered **both** `SocialMedia` and `Hopelessness`, the share reporting hopelessness is **higher** when the social-media risk indicator is **1** than when it is **0** (roughly **44% vs 33%** in this file). That is a **descriptive association**: it does not prove social media *caused* hopelessness—both could be driven by stress, sleep, or other factors not in this two-variable slice. 

**Question 2:** *What is the relationship between cyberbullying and adolescent mental health outcomes?*

*Same approach for Question 2: only students with non-missing answers on **both** cyberbullying and the mental-health outcome are compared.*

In [14]:
# Question 2 — Cyberbullying vs self-reported "not good" mental health (0/1 items).

q2 = df[["CyberBullying", "NotGoodMentalHealth"]].dropna()
print(f"Students who answered BOTH items: {len(q2):,}\n")

tbl2 = q2.groupby("CyberBullying")["NotGoodMentalHealth"].agg(["count", "mean"]).round(3)
tbl2["pct_not_good_1"] = (tbl2["mean"] * 100).round(1)
print("Mean NotGoodMentalHealth by CyberBullying (1 = experienced cyberbullying in this coding):")
print(tbl2)

print("\nRates (NotGoodMentalHealth == 1):")
for code, grp in q2.groupby("CyberBullying"):
    pct = (grp["NotGoodMentalHealth"] == 1).mean() * 100
    label = "experienced cyberbullying" if int(code) == 1 else "did not"
    print(f"  {label}: {pct:.1f}% (n={len(grp):,})")

Students who answered BOTH items: 15,573

Mean NotGoodMentalHealth by CyberBullying (1 = experienced cyberbullying in this coding):
               count   mean  pct_not_good_1
CyberBullying                              
0.0            12835  0.264            26.4
1.0             2738  0.506            50.6

Rates (NotGoodMentalHealth == 1):
  did not: 26.4% (n=12,835)
  experienced cyberbullying: 50.6% (n=2,738)


**Interpretation:** Students who report **cyberbullying** (`CyberBullying == 1`) also report **poor mental health** (`NotGoodMentalHealth == 1`) at a **much higher rate** than students who do not (about **51% vs 26%** in this extract). That pattern matches what we would expect from past research, but this table is still **only a cross-section**: bullying and mental health can reinforce each other over time, and the survey cannot separate direction. 

**Question 3:** *How does sleep relate to social media use and mental health among teenagers?*

In [15]:
# Question 3 — Eight+ hours of sleep vs social media indicator and hopelessness.
# EightOrMoreHoursSleep is 1 if student reports 8+ hours sleep, 0 otherwise (per YRBS-style coding here).

q3 = df[["EightOrMoreHoursSleep", "SocialMedia", "Hopelessness"]].dropna()
print(f"Students with all three items non-missing: {len(q3):,}\n")

print("Mean SocialMedia risk indicator by sleep group (0 = under 8h, 1 = 8h or more):")
print(q3.groupby("EightOrMoreHoursSleep")["SocialMedia"].mean().round(3))
print("\nMean Hopelessness by sleep group:")
print(q3.groupby("EightOrMoreHoursSleep")["Hopelessness"].mean().round(3))

print("\nHopelessness rate (%) by sleep group:")
for code, grp in q3.groupby("EightOrMoreHoursSleep"):
    label = "8+ hours sleep" if int(code) == 1 else "Under 8 hours sleep"
    pct = (grp["Hopelessness"] == 1).mean() * 100
    print(f"  {label}: {pct:.1f}% (n={len(grp):,})")

Students with all three items non-missing: 13,622

Mean SocialMedia risk indicator by sleep group (0 = under 8h, 1 = 8h or more):
EightOrMoreHoursSleep
0.0    0.801
1.0    0.741
Name: SocialMedia, dtype: float64

Mean Hopelessness by sleep group:
EightOrMoreHoursSleep
0.0    0.460
1.0    0.271
Name: Hopelessness, dtype: float64

Hopelessness rate (%) by sleep group:
  Under 8 hours sleep: 46.0% (n=10,538)
  8+ hours sleep: 27.1% (n=3,084)


**Interpretation:** Among teens who answered all three items, those reporting **8 or more hours of sleep** show **lower average hopelessness** than those under eight hours (about **27% vs 46%** reporting hopelessness). The social-media indicator is also **slightly lower** on average in the 8+ sleep group. Together that supports a **sleep–mood** story practitioners care about, but it is still observational: teens who sleep more may differ in many other ways (parent rules, homework load, anxiety).

---

## Section 4 — Visualization

Create **at least one chart per research question** (three charts total below). Each chart should:

- Have a **title that states the finding**, not only the variable names
- Have **labeled axes** (Plotly `labels=` helps)
- Use a chart type that fits the data (**grouped bar** for Chart 1 to compare **two outcomes** at social-media 0 vs 1; **simple bar** for Charts 2–3 for a **single proportion** across two groups)

Run the cells **in order** after `df` is loaded. Each block builds its own small summary table from the same filters used in Section 3.


### Chart 1 — Question 1 (social media, mental health, and hopelessness)

**Grouped bar chart:** at each **social-media indicator** (0 vs 1), two bars compare the **share answering “yes”** on **hopelessness** and on **not-good mental health** (both items are 0/1 here, so means equal proportions). Uses rows with **all three** columns non-missing.


In [ ]:
# --- Chart 1 — Question 1: SocialMedia vs Hopelessness and NotGoodMentalHealth ---
# Grouped bars: compare two 0/1 outcomes at social-media 0 vs 1 (pairwise-complete rows).

q1_plot = df[["SocialMedia", "Hopelessness", "NotGoodMentalHealth"]].dropna()
summary1 = q1_plot.groupby("SocialMedia", as_index=False).agg(
    Hopelessness=("Hopelessness", "mean"),
    NotGoodMentalHealth=("NotGoodMentalHealth", "mean"),
)
long1 = summary1.melt(
    id_vars="SocialMedia",
    value_vars=["Hopelessness", "NotGoodMentalHealth"],
    var_name="Outcome",
    value_name="proportion",
)
long1["Outcome"] = long1["Outcome"].map(
    {
        "Hopelessness": "Hopelessness",
        "NotGoodMentalHealth": "Not good mental health",
    }
)

fig1 = px.bar(
    long1,
    x="SocialMedia",
    y="proportion",
    color="Outcome",
    barmode="group",
    title="Hopelessness and poor mental health by social-media indicator",
    labels={
        "SocialMedia": "Social media indicator (0 vs 1)",
        "proportion": "Proportion answering yes",
        "Outcome": "Outcome",
    },
)

fig1.update_layout(
    yaxis_tickformat=".0%",
    xaxis=dict(tickmode="linear", tick0=0, dtick=1),
    legend_title_text="",
)

fig1.show()


### Chart 2 — Question 2 (cyberbullying and mental health)

In [ ]:
# --- Chart 2 — Question 2: cyberbullying vs self-rated poor mental health ---

q2_plot = df[["CyberBullying", "NotGoodMentalHealth"]].dropna()
summary2 = q2_plot.groupby("CyberBullying", as_index=False)["NotGoodMentalHealth"].mean()
summary2["group"] = summary2["CyberBullying"].map(
    {0.0: "No cyberbullying (0)", 1.0: "Cyberbullying (1)"}
)

fig2 = px.bar(
    summary2,
    x="group",
    y="NotGoodMentalHealth",
    title="Poor mental health is much more common among students who report cyberbullying",
    labels={
        "NotGoodMentalHealth": "Proportion with not-good mental health",
        "group": "",
    },
)
fig2.update_layout(yaxis_tickformat=".0%", showlegend=False)
fig2.show()

### Chart 3 — Question 3 (sleep and hopelessness)

Same filter as the printed tables in Section 3: only students with **both** `EightOrMoreHoursSleep` and `Hopelessness` answered.

In [36]:
# --- Chart 3 — Question 3: sleep (8+ hours) vs hopelessness ---
# Pairwise complete rows: `EightOrMoreHoursSleep` and `Hopelessness` both answered (Section 3 filter).

q3_plot = df[["EightOrMoreHoursSleep", "Hopelessness"]].dropna()
summary3 = q3_plot.groupby("EightOrMoreHoursSleep", as_index=False)["Hopelessness"].mean()
summary3["group"] = summary3["EightOrMoreHoursSleep"].map(
    {0.0: "Under 8 hours sleep", 1.0: "8 or more hours sleep"}
)

fig3 = px.bar(
    summary3,
    x="group",
    y="Hopelessness",
    title="Students who sleep 8+ hours report hopelessness less often",
    labels={"Hopelessness": "Proportion reporting hopelessness", "group": ""},
)
fig3.update_layout(yaxis_tickformat=".0%", showlegend=False)
fig3.show()

**Chart rationale:** **Chart 1** uses **`plotly.express.bar`** with **`barmode="group"`** so readers can compare **two outcomes** (**hopelessness** and **not-good mental health**) **side by side** at **social-media 0 vs 1** without implying a continuous “trend” between the two points (which a line chart can suggest when *x* is categorical). **Charts 2 and 3** each show **one outcome proportion** across **two groups**, so a **simple bar chart** makes the **gap in height** the main story. Titles describe **patterns in the survey**, not causation—many factors co-occur with sleep, bullying, and screen-related risk, and this cross-section cannot prove what would change if one factor changed.

---

## Section 5 — Conclusions

Write 3–5 sentences summarizing what you found. Address these questions:

- What is the most important thing your analysis revealed?
- What surprised you?
- What would you investigate next if you had more time or data?
- What are the limitations of this analysis — what can't you conclude from this data?

Then complete the competency claim below.

**Summary of findings:** In this YRBS-style extract, **social media risk scores line up with hopelessness**: among teens who answered both items, roughly **one in three** in the lower indicator group versus **nearly half** in the higher group report hopelessness. **Cyberbullying** shows an even starker split with **self-rated poor mental health**—about **half** of cyberbullied students versus **about a quarter** of others. **Sleep** also lines up with mood: students reporting **8+ hours of sleep** report hopelessness **less often** than those under eight hours, with a slightly lower social-media risk score on average.

What surprised me most was how **large** the cyberbullying gap is on a single mental-health item, which reinforces why schools invest in climate and safety programs. If I had more time, I would **use survey weights** (`weight`, `stratum`, `psu`) for population estimates, add **plots of distributions** (not only means), and test whether sleep **mediates** part of the social-media association.

**Limitations:** These are **cross-sectional self-reports**; I cannot claim **causation**, and missing answers on some modules mean each analysis uses a **different subset** of students. Variable codes should be checked against the **official codebook** before publishing policy numbers.

---
